### PR TITLE
Set up https://github.com/knative-sandbox/operator

### PR DESCRIPTION
**Pull request by evankanderson**
## What this PR does, why we need it:

This sets up initial prow configuration for https://github.com/knative-sandbox/operator, the location of the new unified operator for Knative, sponsored by the operations WG.

## Special notes to reviewers:

I think I passed the typing test, but please double check.

Also, this is set up in a new org, so there may need to be some additional permissions checks and may be even new tokens provisioned (not sure0.

## User-visible changes in this PR:

* Initial automation and testgrid integration for the unified operator.

## Side note

There seems to be a lot of duplication in the Prow configuration. It might be worth suggesting some form of either defaulting or templating support in Prow to the upstream maintainers. I found myself needing to type the same "agent:Kubernetes\ndecorate: true" line many times, as well as repeating the `name` of the configuration stanza at least 3 times within the block.

### CODE REVIEW

File: config/prow/jobs/config.yaml [hunk 1]
```diff
@@ -4671,6 +4671,105 @@ presubmits:
...
+        - "--github-token=/etc/covbot-token/token"
+        volumeMounts:
+        - name: covbot-token
+          mountPath: /etc/covbot-token
+          readOnly: true
+      volumes:
+      - name: covbot-token
+        secret:
+          secretName: covbot-token
+
```
Comment (mattmoor) [inline config/prow/jobs/config.yaml]: Remove trailing whitespace:
```suggestion

```

**Comment by chizhg** [comment]
It seems you added the config files in a wrong way, the only file you need to make changes is `config/prow/config_knative.yaml`, and then all other config files will be auto-generated after running `./hack/generate_configs.sh`, see more details in https://github.com/knative/test-infra/blob/master/guides/prow_setup.md (the doc is a bit outdated though, I will make the improvements soon)

**Comment by evankanderson** [comment]
From #1840, I realized that I needed to run `./hack/update-codegen.sh`, which made slightly different changes than I'd done by hand.

**Comment by evankanderson** [comment]
Also, `./hack/update-codegen.sh` actually fails for me, so I ran `./hack/generate-configs.sh` instead:

```shell
$ ./hack/update-codegen.sh 
/Users/evana/go/src/knative.dev/test-infra/hack/update-deps.sh: line 44: DEP_FLAGS[@]: unbound variable
```

File: config/prow/config_knative.yaml [hunk 2]
```diff
@@ -965,3 +975,17 @@ periodics:
...
       go113: true
+
+  knative-sandbox/operator:
+    - continuous: true
+      dot-dev: true
+      go113: true
+    - nightly: true
+      dot-dev: true
+      go113: true
+    - dot-release: true
```
Comment (chizhg) [inline config/prow/config_knative.yaml]: We need a hack to set the org name as env var for release jobs for non-knative projects, see https://github.com/knative/test-infra/blob/master/config/prow/config_knative.yaml#L906. 
So please add below lines to these two job settings:
```
env-vars:
    - ORG_NAME=knative-sandbox
```

File: config/prow/config_knative.yaml [hunk 3]
```diff
@@ -402,6 +402,16 @@ presubmits:
     - integration-tests: false
     - go-coverage: false
 
+  knative-sandbox/operator:
+    - build-tests: true
+      dot-dev: true
```
Comment (chizhg) [inline config/prow/config_knative.yaml]: Do projects in this org also use the `knative.dev` path alias for dep? If not, please remove all `dot-dev: true` settings.

**Comment by chizhg** [comment]
> Also, `./hack/update-codegen.sh` actually fails for me, so I ran `./hack/generate-configs.sh` instead:
> 
> ```shell
> $ ./hack/update-codegen.sh 
> /Users/evana/go/src/knative.dev/test-infra/hack/update-deps.sh: line 44: DEP_FLAGS[@]: unbound variable
> ```

Hmm, I'm running it on MAC but do not see the error. Which system are you using?

**Comment by evankanderson** [comment]
I'm using the following `bash`:

```shell
$ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
Copyright (C) 2007 Free Software Foundation, Inc.
```

File: config/prow/config_knative.yaml (hunk 3 above)
Comment (evankanderson) [reply config/prow/config_knative.yaml]: We haven't figured this out yet, so I'll remove it.

File: config/prow/config_knative.yaml (hunk 2 above)
Comment (evankanderson) [reply config/prow/config_knative.yaml]: So this is only for `-release` steps, correct?

Updated.

**Comment by chizhg** [comment]
> I'm using the following `bash`:
> 
> ```shell
> $ bash --version
> GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
> Copyright (C) 2007 Free Software Foundation, Inc.
> ```

Thanks, I have created https://github.com/knative/test-infra/issues/1856 for tracking.

**Comment by chizhg** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**